openedx/core/djangoapps/bookmarks/views.py

```diff
@@ -97,9 +96,35 @@ def error_response(self, developer_message, user_message=None, error_status=stat
         )
 
 
-@method_decorator(name='get', decorator=swagger_auto_schema(
-    operation_summary="Get a paginated list of bookmarks for a user.",
-    operation_description=u"""
+class BookmarksListView(ListCreateAPIView, BookmarksViewMixin):
+    """REST endpoints for lists of bookmarks."""
+
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    pagination_class = BookmarksPagination
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = BookmarkSerializer
+
+    @swagger_auto_schema(
+        manual_parameters=[
+            openapi.Parameter(
+                'course_id',
+                openapi.IN_QUERY,
+                type=openapi.TYPE_STRING,
+                description="The id of the course to limit the list",
+            ),
+            openapi.Parameter(
+                'fields',
+                openapi.IN_QUERY,
+                type=openapi.TYPE_STRING,
+                description="""
+                    The fields to return: display_name, path.
+                    """,
+            ),
+        ],
+    )
+    def get(self, request, *args, **kwargs):
+        """Get a paginated list of bookmarks for a user.
```

Comment:
Nit: edx-platform convention is to put the first line of the docstring on a new line. Fortunately, both styles work with your swagger_auto_schema code

Reply:
:+1: